src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/SetPosition.java

```diff
@@ -29,9 +32,18 @@ public SetPosition(ConnectionImpl connection) {
 		}
 		//PE sends position that intersects blocks bounding boxes in some cases
 		//Server doesn't accept such movements and will send a set position, but we ignore it unless it is above leniency
-		if (cache.getMovementCache().isPEPositionAboveLeniency()) {
+		if (movecache.isPEPositionAboveLeniency()) {
 			packets.add(create(cache.getWatchedEntityCache().getSelfPlayer(), x, y + 0.01, z, pitch, yaw, ANIMATION_MODE_TELEPORT));
 		}
+		movecache.setPEClientPosition(x, y, z);
```

Comment:
Do you think its a good idea to send this on every player teleport / move?
While its true we need the thingy render thing pointed to the players current location so the chunks don't render everywhere. I think its better to send them with each chunk, so when a new chunk is send the renderposition thing is resend, that way we also don't have to worry about the client not rendering chunks if they walk themselves and the server doesn't teleport.

Reply:
Since you already seem to be sending the NETWORK_CHUNK_PUBLISHER_UPDATE_PACKET inside the chunk packet, can you try just removing this code from the position packet?

Reply:
was just looking through this again.... i cant remember exactly why i put this here 🤷‍♂️ ill just keep it with the chunk update, should be fine i think

Reply:
hmmm think removing this may have caused some issues with initial spawn. ill see if i can add it to the respawn packet or something